test/rest_api/test_repair_task.py

```diff
@@ -0,0 +1,68 @@
+import sys
+
+# Use the util.py library from ../cql-pytest:
+sys.path.insert(1, sys.path[0] + '/../cql-pytest')
+from util import new_test_table, new_test_keyspace
+from rest_util import set_tmp_task_ttl
+from task_manager_utils import list_tasks, drain_module_tasks, get_task_status, check_child_parent_relationship
+
+empty_id = "00000000-0000-0000-0000-000000000000"
+module_name = "repair"
+long_time = 1000000000
+
+def run_repair_and_wait(rest_api, keyspace):
+    resp = rest_api.send("POST", f"storage_service/repair_async/{keyspace}")
+    resp.raise_for_status()
+    sequence_number = resp.json()
+    resp = rest_api.send("GET", f"storage_service/repair_status", { "id": sequence_number })
+    resp.raise_for_status()
+    return sequence_number
+
+def test_repair_task_progress(cql, this_dc, rest_api):
+    drain_module_tasks(rest_api, module_name)
```

Comment:
What does this do exactly? 

I saw

```
def drain_module_tasks(rest_api, module_name):
    tasks = [task for task in list_tasks(rest_api, module_name, True)]
    for task in tasks:
        resp = rest_api.send("GET", f"task_manager/wait_task/{task['task_id']}")
        # The task may be already unregistered.
        assert resp.status_code == requests.codes.ok or resp.status_code == requests.codes.bad_request, "Invalid status code"
```

This does not seem to wait for the pending tasks to finish.

Reply:
Different tests are run on the same scylla instance and this test shouldn't check all the leftovers in task manager (by default each task stays there for 10s after it is complete).

`task_manager/wait_task` does two things: waits for a task to finish and unregisters it.

Reply:
But drain_module_tasks goes over all the tasks under the module_name (I assume it is repair here). And it waits for the repair tasks to finish, no?

Reply:
Yes, it does